accounts/views.py

```diff
@@ -73,6 +76,81 @@ def citesphere_token(request):
         token.save()
         return Response(status=status.HTTP_201_CREATED)
 
+class ForgotPasswordView(APIView):
+    def post(self, request):
+        username = request.data.get('username', None)
+        if username:
+            try:
+                user = VogonUser.objects.get(username=username)
+            except VogonUser.DoesNotExist:
+                return Response({ "success": False, "message": "User not found" }, status=404)
+            token = jwt.encode({ 
+                'exp': arrow.utcnow().shift(days=1).timestamp,
+                'username': user.username,
+            }, settings.SECRET_KEY, algorithm='HS256').decode('UTF-8')
+            reset_token = ResetToken(user=user, token=token)
+            reset_token.save()
+
+            # Email token
+            send_mail(
+                'Vogon password reset',
+                f'Reset link: http://localhost:8080/reset-password/{token}',
+                'test@vogonweb.com',
+                [user.email],
+                fail_silently=True,
+            )
+
+            return Response({ "success": True })
+        return Response({ "success": False, "message": "Username not specified" }, status=400)
+
+class ResetPasswordView(APIView):
+    def post(self, request):
+        username = request.data.get('username', None)
```

Comment:
I think we should use a serializer for this. That way we can validate the data

Reply:
Done